src/components/ModalWithForm/LoginModal/LoginModal.jsx

```diff
@@ -0,0 +1,95 @@
+import { useState } from "react";
+import "./LoginModal.css";
+import ModalWithForm from "../ModalWithForm";
+
+const LoginModal = ({
+  handleLogin,
+  isOpen,
+  closeActiveModal,
+  handleSignup,
+  handleOutsideClick,
+}) => {
+  const [data, setData] = useState({
+    email: "",
+    password: "",
+  });
+
+  const [errorMessage, setErrorMessage] = useState(""); // For error handling
+  const [isLoading, setIsLoading] = useState(false); // For loading state
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setData((prevData) => ({
+      ...prevData,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setIsLoading(true); // Set loading state when request is sent
+    setErrorMessage('');
+
+    try {
+      await handleLogin(data);  // Call the handleLogin function from props
+      setIsLoading(false);
```

Comment:
` setIsLoading(false);` should be in `finally` not to duplicate it in `then` and `catch`